app/src/main/java/com/everest/moviedb/MovieViewModel.kt

```diff
@@ -0,0 +1,66 @@
+package com.everest.moviedb
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.everest.moviedb.model.APIResponse
+import com.everest.moviedb.model.Movie
+import com.everest.moviedb.model.MovieRepository
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class MovieViewModel : ViewModel() {
+    private val _movieList = MutableLiveData(listOf(Movie(0, "", "", "", "", "", "", "")))
+    val errorMessage = MutableLiveData<String>()
+    val movieList: LiveData<List<Movie>> = _movieList
+
+    private val _currentMovieList = MutableLiveData(listOf(Movie(0, "", "", "", "", "", "", "")))
+    val currentMovieList: LiveData<List<Movie>> = _currentMovieList
+
+    private val _searchMovieList = MutableLiveData(listOf(Movie(0, "", "", "", "", "", "", "")))
+    val searchMovieList: LiveData<List<Movie>> = _searchMovieList
+
+
+
+    fun getAllMovies(movieRepository: MovieRepository) {
+        val response = movieRepository.getAllMovies()
+        response.enqueue(object : Callback<APIResponse> {
```

Comment:
Move the enqueue to repository
Repository should get the data and convert response into UI model and pass it back to viewmodel. So viewmodel will decide what to do with data

check sealed classes how we can return data and handle it easily